### PR TITLE
Organizations

### PR DESCRIPTION
**Pull request by abstractcoder**
Right now adding a project to an Organization does not make it globally visible. I feel that the logic for project visibility is already quite complicated and way may want to consider a simpler approach (such has featuring Organizations instead of projects OR rethinking publish, unpublished, and featured within the context of an Organization).

I also started building in a few abstractions like creating plugs for getting resources, a universal authorization plug, and helper functions to make testing permissions within controller tests easier. If these approaches seem sound I would plan to do a refactor the other controls once we push this update live.

### CODE REVIEW

**Comment by abstractcoder** [comment]
I pushed an update that implemented the feedback today's meeting.

I'm going to make a change where organization members can see unpublished organization projects, and non-members can't. That way organization members can still see all the projects that are part of the organization.

I'm planning to expand and clean up some of the tests as well.

File: lib/digital_public_works/organizations/organization_project.ex [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  schema "organizations_projects" do
+    belongs_to :organization, Organization
+    belongs_to :project, Project
+
+    timestamps()
+  end
+
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> Ecto.Changeset.cast(params, [:organization_id, :project_id])
```
Comment (nickgartmann) [inline lib/digital_public_works/organizations/organization_project.ex]: Any particular reason for not importing Changeset here?

File: lib/digital_public_works/projects.ex [hunk 2]
```diff
@@ -18,65 +18,68 @@ defmodule DigitalPublicWorks.Projects do
...
-    |> Repo.all()
-  end
+  def list_published_projects(search \\ nil), do: all() |> published() |> fetch(search)
 
-  def list_projects(%User{is_admin: true}, search) do
-    Project
-    |> filter_projects(search)
-    |> Repo.all()
-  end
+  def list_featured_projects(), do: all() |> featured() |> fetch()
```
Comment (nickgartmann) [inline lib/digital_public_works/projects.ex:25]: Loving how concise and expressive this got. Main interface is to use the predefined methods to get the specific group of projects you want, and those are easy to add to by composing the queries together. Very nice.

**Review by nickgartmann — APPROVED** [review]
I really love how this feature came together. The transition to plugs cleaned up the controllers quite a bit, the composable queries seemed to be really powerful for building the different ways you can ask for project lists, and the visibility seems clearer now.

The design elements that got added are really nice, the checkbox on the search and "Endorsed by" parts look really nice.

This looks ready to ship to me

File: lib/digital_public_works/organizations/organization_project.ex (hunk 1 above)
Comment (abstractcoder) [reply lib/digital_public_works/organizations/organization_project.ex]: Nope, thanks for pointing this out! I cleaned this up in the affected Ecto schemas.

**Pull request merged**